### ISSUE TITLE
Openssl installation fails using Linux Brew

### ISSUE THREAD

**Issue by sudhirj** [Microsoft/BashOnWindows#1409]
I'm installing packages using the Linux Brew package manager http://linuxbrew.sh/, and reporting errors I see when installing common packages used for development. This is the error for OpenSSL.

* Expected results: OpenSSL should install cleanly.

* Actual results (with terminal output if applicable)

```
Enable-WindowsOptionalFeature@SKULL:~$ brew install openssl
sudo: unable to resolve host SKULL
==> Downloading https://www.openssl.org/source/openssl-1.0.2j.tar.gz
Already downloaded: /home/Enable-WindowsOptionalFeature/.cache/Homebrew/openssl-1.0.2j.tar.gz
==> perl ./Configure --prefix=/home/Enable-WindowsOptionalFeature/.linuxbrew/Cellar/openssl/1.0.2j --openssldir=/home/Enable-WindowsOpt
==> make depend
Last 15 lines from /home/Enable-WindowsOptionalFeature/.cache/Homebrew/Logs/openssl/02.make:
2016-11-23 10:56:49 +0530

make
depend

making depend in crypto...
make[1]: Entering directory `/tmp/openssl-20161123-18599-nr7cxc/openssl-1.0.2j/crypto'
/home/Enable-WindowsOptionalFeature/.linuxbrew/bin/makedepend: 1: /home/Enable-WindowsOptionalFeature/.linuxbrew/bin/makedepend: ELF: not found
/home/Enable-WindowsOptionalFeature/.linuxbrew/bin/makedepend: 3: /home/Enable-WindowsOptionalFeature/.linuxbrew/bin/makedepend: Syntax error: "(" unexpected
mv: cannot stat 'Makefile.new': No such file or directory
make[1]: *** [local_depend] Error 2
make[1]: Leaving directory `/tmp/openssl-20161123-18599-nr7cxc/openssl-1.0.2j/crypto'
make: *** [depend] Error 1
```

* Your Windows build number: 14393.447

* Steps / All commands required to reproduce the error from a brand new installation 
```
ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install)"
PATH="$HOME/.linuxbrew/bin:$PATH"
brew install openssl
```

* Strace of the failing command
```
make[1]: Entering directory `/tmp/openssl-20161123-18599-nr7cxc/openssl-1.0.2j/crypto'
/home/Enable-WindowsOptionalFeature/.linuxbrew/bin/makedepend: 1: /home/Enable-WindowsOptionalFeature/.linuxbrew/bin/makedepend: ELF: not found
/home/Enable-WindowsOptionalFeature/.linuxbrew/bin/makedepend: 3: /home/Enable-WindowsOptionalFeature/.linuxbrew/bin/makedepend: Syntax error: "(" unexpected
mv: cannot stat 'Makefile.new': No such file or directory
make[1]: *** [local_depend] Error 2
make[1]: Leaving directory `/tmp/openssl-20161123-18599-nr7cxc/openssl-1.0.2j/crypto'
make: *** [depend] Error 1
```

* Required packages and commands to install

See [our contributing instructions](/CONTRIBUTING.md) for assistance.

**Comment by sudhirj** [comment]
I'd also suggest using the Linux Brew manager as a test suite of sorts - clean installs of `openssl rbenv ruby-build go git mercurial make gcc` etc will go a long way in ensuring full compatibility.

**Comment by jackchammons** [comment]
I am able to reproduce this error on 14961

**Comment by alexanderwhatley** [comment]
Duplicate of https://github.com/Microsoft/BashOnWindows/issues/471

**Comment by sudhirj** [comment]
Both #471 and this one seems to be related to OpenSSL, but that looks like and issue with an `.so` file not working as expected. At first glance this looked like either a file encoding problem or a missing syscall.

**Comment by yorickdowne** [comment]
Might be a missing syscall. Rakudo * brew has a similar issue with a .so not loading. In that case it's related to execstack, see the workaround at https://github.com/Microsoft/BashOnWindows/issues/283 . If you build this from source with LDFLAGS = @ldflags@ -Wl,-z,noexecstack in makefile.in (or the OpenSSL equivalent to disable execstack), how does it behave?

**Comment by yorickdowne** [comment]
I get `/home/tbehrens/.linuxbrew/bin/makedepend: 1: /home/tbehrens/.linuxbrew/bin/makedepend: ELF: not found` which would point to the issue in https://github.com/Microsoft/BashOnWindows/issues/471, as in "As far as I can tell, the Windows kernel is unable to load ELF executables that have been modified using patchelf." This is fixinbound as of 12/5/2016, so it's quite possible this issue resolves at the same time as 471.

**Comment by sudhirj** [comment]
This should now be marked as fixed, based on #471. Looks like the problem was that WSL didn't correctly run programs touched by `patchelf`. Fixed in Windows Insider Build 15002.
